presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/DefaultFormatFunction.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+public interface DefaultFormatFunction
```

Comment:
this one is never used in practice, as whenever you take something out of `formatFunctions` you cast it to `JsonDateTimeFormatFunction`. Is that a bug?
Also I do not huge fan of idea of having format functions which do not satisfy common interface in single list.

Reply:
I needed a filler function for non-temporal columns. I agree this is less than ideal, I'm not sure what to replace it with. This function is never actually used in the encoder. I was thinking that in the future we could replace all the `appendLong, appendBoolean...` methods with a precompiled list of lambda functions to format/add the column value to the encoder. For now we need to figure out a different solution. I could change the `List<JsonFormatFunctions> formatFunctions` field in `JsonRowEncoder` to a `Map` and have the column name as the key? That way I don't need to access it with the `currentColumnIndex`.

Reply:
I find the approach with default function cumbersome. Maybe instead having a hierarchy here we can just have as set of functions and single class with methods delegating to functors. Then we can provide the fuctions we care about as lambdas via builder. The ones which are not explicitly set would just throw exception. 

Sth like this:
```java
public class JsonFormatFunction
{
    private final Function<Instant, String> formatTimestampLegacyFunc;
    private final Function<LocalDateTime, String> formatTimestampFunc;
    //private final Function<Something, String> formatSomething;

    public JsonFormatFunction(Function<Instant, String> formatTimestampLegacyFunc, Function<LocalDateTime, String> formatTimestampFunc)
    {
        this.formatTimestampLegacyFunc = formatTimestampLegacyFunc;
        this.formatTimestampFunc = formatTimestampFunc;
    }

    public String formatTimestampLegacy(Instant value)
    {
        return formatTimestampLegacyFunc.apply(value);
    };

    public String formatTimestamp(LocalDateTime value)
    {
        return formatTimestampFunc.apply(value);
    }

    public static class Builder
    {
        private Function<Instant, String> formatTimestampLegacyFunc = (ignored) -> { throw new RuntimeException("not supported argument type"); };
        private Function<LocalDateTime, String> formatTimestampFunc = (ignored) -> { throw new RuntimeException("not supported argument type"); };

        JsonFormatFunction build()
        {
            return new JsonFormatFunction(
                    formatTimestampLegacyFunc,
                    formatTimestampFunc);
        }

        public Builder setFormatTimestampLegacyFunc(Function<Instant, String> formatTimestampLegacyFunc)
        {
            this.formatTimestampLegacyFunc = formatTimestampLegacyFunc;
            return this;
        }

        public Builder setFormatTimestampFunc(Function<LocalDateTime, String> formatTimestampFunc)
        {
            this.formatTimestampFunc = formatTimestampFunc;
            return this;
        }
    }
}
```

One downside of this is that is a bit verbose. It also involves extra boxing/unboxing - not sure how much of a problem that is with new JVMs.